lib/ansible/plugins/vars/sops_vars.py

```diff
@@ -0,0 +1,258 @@
+from ansible.utils.display import Display
+display = Display()
+
+FOUND = {}
+DEFAULT_VALID_EXTENSIONS = [".sops.yaml", ".sops.yml", ".sops.json"]
+
+# From https://github.com/mozilla/sops/blob/master/cmd/sops/codes/codes.go
+# Should be manually updated
+sops_error_codes = {
+    1: "SopsErrorGeneric",
+    2: "SopsCouldNotReadInputFile",
+    3: "SopsCouldNotWriteOutputFile",
+    4: "SopsErrorDumpingTree",
+    5: "SopsErrorReadingConfig",
+    6: "SopsErrorInvalidKMSEncryptionContextFormat",
+    7: "SopsErrorInvalidSetFormat",
+    8: "SopsErrorConflictingParameters",
+    21: "SopsErrorEncryptingMac",
+    23: "SopsErrorEncryptingTree",
+    24: "SopsErrorDecryptingMac",
+    25: "SopsErrorDecryptingTree",
+    49: "SopsCannotChangeKeysFromNonExistentFile",
+    51: "SopsMacMismatch",
+    52: "SopsMacNotFound",
+    61: "SopsConfigFileNotFound",
+    85: "SopsKeyboardInterrupt",
+    91: "SopsInvalidTreePathFormat",
+    100: "SopsNoFileSpecified",
+    128: "SopsCouldNotRetrieveKey",
+    111: "SopsNoEncryptionKeyFound",
+    200: "SopsFileHasNotBeenModified",
+    201: "SopsNoEditorFound",
+    202: "SopsFailedToCompareVersions",
+    203: "SopsFileAlreadyEncrypted"
+}
+
+
+class SopsError(AnsibleError):
+    ''' extend AnsibleError class with sops specific informations '''
+
+    def __init__(self, filename, exit_code, message="Unknown error",):
+        message = "error with file %s: sops exited with code %d: %s" % (filename, exit_code, message)
+        super(SopsError, self).__init__(message=message)
+
+
+class SopsErrorGeneric(SopsError):
+    pass
+
+
+class SopsCouldNotReadInputFile(SopsError):
+    pass
+
+
+class SopsCouldNotWriteOutputFile(SopsError):
+    pass
+
+
+class SopsErrorDumpingTree(SopsError):
+    pass
+
+
+class SopsErrorReadingConfig(SopsError):
+    pass
+
+
+class SopsErrorInvalidKMSEncryptionContextFormat(SopsError):
+    pass
+
+
+class SopsErrorInvalidSetFormat(SopsError):
+    pass
+
+
+class SopsErrorConflictingParameters(SopsError):
+    pass
+
+
+class SopsErrorEncryptingMac(SopsError):
+    pass
+
+
+class SopsErrorEncryptingTree(SopsError):
+    pass
+
+
+class SopsErrorDecryptingMac(SopsError):
+    pass
+
+
+class SopsErrorDecryptingTree(SopsError):
+    pass
+
+
+class SopsCannotChangeKeysFromNonExistentFile(SopsError):
+    pass
+
+
+class SopsMacMismatch(SopsError):
+    pass
+
+
+class SopsMacNotFound(SopsError):
+    pass
+
+
+class SopsConfigFileNotFound(SopsError):
+    pass
+
+
+class SopsKeyboardInterrupt(SopsError):
+    pass
+
+
+class SopsInvalidTreePathFormat(SopsError):
+    pass
+
+
+class SopsNoFileSpecified(SopsError):
+    pass
+
+
+class SopsCouldNotRetrieveKey(SopsError):
+    pass
+
+
+class SopsNoEncryptionKeyFound(SopsError):
+    pass
+
+
+class SopsFileHasNotBeenModified(SopsError):
+    pass
+
+
+class SopsNoEditorFound(SopsError):
+    pass
+
+
+class SopsFailedToCompareVersions(SopsError):
+    pass
+
+
+class SopsFileAlreadyEncrypte(SopsError):
```

Comment:
https://github.com/mozilla/sops/blob/afd073a5be0fe2232d7cd345b9b30edc70ccb962/cmd/sops/encrypt.go#L32
```suggestion
class SopsFileAlreadyEncrypted(SopsError):
```

Reply:
@endorama ☝️